### DISCUSSION TITLE
Pull requests need performance optimization

### DISCUSSION THREAD

**Discussion by spmonahan** [community/community#111001] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

Interacting with the UI on this PR is unreasonably slow: https://github.com/microsoft/fluentui/pull/30689/files

All interactions have a several second delay and filtering (to only show `.ts` and `.tsx` files for example) does nothing to alleviate the issue 😬

Opening DevTools and running:

```js
document.querySelectorAll('*').length
```

Returns `103458`. As a point of reference, [Lighthouse classifies ~1,500 DOM nodes as excessive](https://developer.chrome.com/docs/lighthouse/performance/dom-size/).

The above PR is certainly on the larger side but the issue starts to appear, though not nearly so severely, on much more modest PRs: https://github.com/microsoft/fluentui/pull/30633/files (this page as 25,028 DOM nodes by my count).

I'd like to see the PR files page become more optimized. Perhaps virtualization for the file tree view and for the list of diffed files.

I'm using MS Edge Version 122.0.2365.66 (Official build) (64-bit) on Windows 11 on a Surface Book 3.

**Comment by greglittlefield-wf** [comment]
I've run into the same issue with even worse performance for larger PRs, with the page locking up for several seconds after certain interactions (clicking a checkbox to mark a file as viewed, clicking on a line to create a new comment, etc.).

I noticed that while the performance issues are pretty severe in Chrome, they're almost nonexistent Firefox.

I tried doing some performance profiling in Chrome, and there were Long Tasks that weren't explained by anything, so perhaps it's something in Chrome's internals that's slow?

**Reply by spmonahan** [reply]
I've noticed the same thing about Firefox handling these large PRs no problem.

**Comment by uttambodara** [comment]
Response #1
